tests/test_modelgen.py

```diff
@@ -43,304 +43,23 @@ def get_default():
-
-    # Tests for DeepconvLSTM model:
-    def test_deepconvlstm_batchnorm_dim(self):
-        """The output shape of the batchnorm should be (None, nr_timesteps, nr_channels, nr_filters)"""
-        model_type = ConvLSTM((None, 20, 3), 2)
-        model = model_type.create_model(**{"filters": [32, 32],
-                                           "lstm_dims": [32, 32]})
-
-        batchnormlay = model.layers[3]
-        assert batchnormlay.output_shape == (None, 20, 3, 32)
-
-    def test_deepconvlstm_enough_batchnorm(self):
-        """LSTM model should contain as many batch norm layers as it has activations layers"""
-        model_type = ConvLSTM((None, 20, 3), 2)
-        model = model_type.create_model(**{"filters": [32, 32, 32],
-                                           "lstm_dims": [32, 32, 32]})
-
-        batch_norm_layers = len([l for l in model.layers if 'BatchNormalization' in str(l)])
-        activation_layers = len([l for l in model.layers if 'Activation' in str(l)])
-        assert batch_norm_layers == activation_layers
-
-    def test_DeepConvLSTM_hyperparameters_nrconvlayers(self):
-        """ Number of Conv layers from range [4, 4] should be 4. """
-        custom_settings = get_default()
-        kwargs = {'deepconvlstm_min_conv_layers': 4,
-                  'deepconvlstm_max_conv_layers': 4}
-        # Replace default parameters with input
-        for key, value in kwargs.items():
-            if key in custom_settings:
-                custom_settings[key] = value
-
-        model_type = ConvLSTM(None, None, **custom_settings)
-        hyperparams = model_type.generate_hyperparameters()
-
-        assert len(hyperparams.get('filters')) == 4
-
-    def test_deepconvlstm_starts_with_batchnorm(self):
-        """ DeepConvLSTM models should always start with a batch normalization layer. """
-        model_type = ConvLSTM((None, 20, 3), 2)
-        model = model_type.create_model(**{"filters": [32, 32],
-                                           "lstm_dims": [32, 32]})
-
-        assert 'BatchNormalization' in str(type(model.layers[0])), 'Wrong layer type.'
-
-    # Tests for ResNet model:
-    def test_ResNet_starts_with_batchnorm(self):
-        """ ResNet models should always start with a batch normalization layer. """
-        model_type = ResNet((None, 20, 3), 2)
-        model = model_type.create_model(16, 20)
-
-        assert 'BatchNormalization' in str(type(model.layers[1])), 'Wrong layer type.'
-
-
-    def test_ResNet_first_sandwich_layers(self):
-        """ ResNet models should always start with a residual module. """
-        model_type = ResNet((None, 20, 3), 2)
-        model = model_type.create_model(16, 20)
 
-        assert 'Conv1D' or 'Convolution1D' in str(type(model.layers[2])), 'Wrong layer type.'
-        assert 'BatchNormalization' in str(type(model.layers[3])), 'Wrong layer type.'
-        assert 'ReLU' in str(type(model.layers[4])), 'Wrong layer type.'
+# TODO: Move this to an utils file?
```

Comment:
please don't add TODO's in the code. Just make an issue in the issue tracker instead.
But indeed, it could be moved. As it's test specific, it should go in the test package and not in the production code. This would be a good place I think: 
https://github.com/NLeSC/mcfly/blob/master/tests/test_tools.py
Do you agree?

Reply:
Done, issue #263